### PR TITLE
Add Rubyfang OC

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by Memeji — APPROVED** [review]

**Comment by FoxxoTrystan** [comment]
Il verify it and test it tomorrow, im sleepy!

File: Content.Server/Polymorph/Components/PolymorphProviderComponent.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+using Content.Shared.Polymorph;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Polymorph.Components;
+
+[RegisterComponent]
+[Access(typeof(PolymorphSystem))]
+public sealed partial class PolymorphProviderComponent  : Component
+{
+    [DataField(required: true)]
```
Comment (FoxxoTrystan) [inline Content.Server/Polymorph/Components/PolymorphProviderComponent.cs:11]:
```suggestion
    [DataField]
```

File: Content.Server/Polymorph/Systems/PolymorphSystem.Provider.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+using Content.Shared.Inventory.Events;
+using Content.Shared.Polymorph;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Polymorph.Systems;
+
+public sealed partial class PolymorphSystem
+{
+    [Dependency] private readonly IEntityManager _entities = default!;
```
Comment (FoxxoTrystan) [inline Content.Server/Polymorph/Systems/PolymorphSystem.Provider.cs:13]:
```suggestion
```

File: Content.Server/Polymorph/Systems/PolymorphSystem.Provider.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        SubscribeLocalEvent<PolymorphProviderComponent, GotEquippedEvent>(OnEquipped);
+        SubscribeLocalEvent<PolymorphProviderComponent, GotUnequippedEvent>(OnUnEquipped);
+
+    }
+
+    private void OnUnEquipped(EntityUid uid, PolymorphProviderComponent component, ref GotUnequippedEvent args)
+    {
+        var polymorphable = _entities.EnsureComponent<PolymorphableComponent>(args.Equipee);
+        RemovePolymorphAction(component.Polymorph, (args.Equipee, polymorphable));
```
Comment (FoxxoTrystan) [inline Content.Server/Polymorph/Systems/PolymorphSystem.Provider.cs:24]:
```suggestion
        if (TryComp<PolymorphableComponent>(args.Equipee, out var polymorphable))
            RemovePolymorphAction(component.Polymorph, (args.Equipee, polymorphable));
```

File: Content.Server/Polymorph/Systems/PolymorphSystem.Provider.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    private void OnEquipped(EntityUid uid, PolymorphProviderComponent component, ref GotEquippedEvent args)
+    {
+
+        var polymorphable = _entities.EnsureComponent<PolymorphableComponent>(args.Equipee);
```
Comment (FoxxoTrystan) [inline Content.Server/Polymorph/Systems/PolymorphSystem.Provider.cs:30]:
```suggestion
        var polymorphable = EnsureComp<PolymorphableComponent>(args.Equipee);
```

**Review by ShadesMars — CHANGES_REQUESTED** [review]
All seem good in general except some code using outated method, suggestions provide the update ones.

Also OnUnEquipped Ensure the comp to check? instead use TryComp witch check if the user has the comp.

File: Content.Server/Polymorph/Components/PolymorphProviderComponent.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+using Content.Server.Polymorph.Systems;
+using Content.Shared.Polymorph;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Polymorph.Components;
+
+[RegisterComponent]
+[Access(typeof(PolymorphSystem))]
+public sealed partial class PolymorphProviderComponent  : Component
```
Comment (FoxxoTrystan) [inline Content.Server/Polymorph/Components/PolymorphProviderComponent.cs:9]:
```suggestion
public sealed partial class PolymorphProviderComponent : Component
```

File: Content.Server/Polymorph/Systems/PolymorphSystem.Provider.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+namespace Content.Server.Polymorph.Systems;
+
+public sealed partial class PolymorphSystem
+{
+    [Dependency] private readonly IEntityManager _entities = default!;
+    private void InitializeProvider()
+    {
+        SubscribeLocalEvent<PolymorphProviderComponent, GotEquippedEvent>(OnEquipped);
+        SubscribeLocalEvent<PolymorphProviderComponent, GotUnequippedEvent>(OnUnEquipped);
+
```
Comment (FoxxoTrystan) [inline Content.Server/Polymorph/Systems/PolymorphSystem.Provider.cs:18]:
```suggestion
```

File: Content.Server/Polymorph/Components/PolymorphProviderComponent.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace Content.Server.Polymorph.Components;
+
+[RegisterComponent]
+[Access(typeof(PolymorphSystem))]
+public sealed partial class PolymorphProviderComponent : Component
+{
+    [DataField]
+    public ProtoId<PolymorphPrototype> Polymorph;
+
```
Comment (FoxxoTrystan) [inline Content.Server/Polymorph/Components/PolymorphProviderComponent.cs:13]:
```suggestion
```

**Review by ShadesMars — APPROVED** [review]

**Comment by FoxxoTrystan** [comment]
Added to V7.

**Pull request merged**